piff/optical_model.py

```diff
@@ -0,0 +1,163 @@
+
+# Copyright (c) 2016 by Mike Jarvis and the other collaborators on GitHub at
+# https://github.com/rmjarvis/Piff  All rights reserved.
+#
+# Piff is free software: Redistribution and use in source and binary forms
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+
+"""
+.. module:: optical_model
+"""
+
+from __future__ import print_function
+
+import galsim
+import fitsio
+import numpy as np
+
+from .model import Model
+from .star import Star, StarFit, StarData
+
+des_pupil_template = {'obscuration': 0.301 / 0.7174,
+                      'nstruts': 4,
+                      # aaron plays between 19 mm thick and 50 mm thick
+                      'strut_thick': 0.050 * (1462.526 / 4010.) / 2.0, # conversion factor is nebulous?!
+                      'strut_angle': 45 * galsim.degrees}
+
+class Optical(Model):
+    def __init__(self, rzero=0.1, sigma=0., g1=0., g2=0., pupil_path='None', lam=500., diam=4.274419, optical_template='des', **kwargs):
+        """Initialize the Optical Model
+
+        :param rzero:               Atmospheric seeing. Usually in the 0.1 - 0.2 range.
+        :param g1, g2:              Shear to apply to final image. Simulates vibrational modes.
+        :param sigma:               Convolve with gaussian of size sigma.
+        :param pupil_path:          If a path is given, load up a pupil image, else
+                                    make image from galsim parameters referencing optical_template
```

Comment:
Also, the default should be listed here in the doc string.  (Likewise for the rest of these that are missing them.)  The notation I've been trying to keep to is `[default: value]`, which matches what we decided to do for GalSim (which makes it easier for me to remember if we use the same syntax here).